kyaml/yaml/rnode.go

```diff
@@ -106,13 +109,46 @@ func NewScalarRNode(value string) *RNode {
 
 // NewStringRNode returns a new Scalar *RNode containing the provided string.
 // If the string is non-utf8, it will be base64 encoded, and the tag
-// will indicate binary data.
-func NewStringRNode(value string) *RNode {
+// will indicate binary data. Otherwise, the tag will be String.
+func NewStringRNode(s string) *RNode {
 	n := yaml.Node{Kind: yaml.ScalarNode}
-	n.SetString(value)
+	n.SetString(s)
+	if n.Tag == "" {
```

Comment:
It seems the n.Tag is always set so this if condition will never be triggered. 
See [ln485](https://github.com/mengqiy/kustomize/blob/95c5b686bea0d10c2de689fdb510972ef583f3ac/kyaml/internal/forked/github.com/go-yaml/yaml/yaml.go#L485)
[ln488](https://github.com/mengqiy/kustomize/blob/95c5b686bea0d10c2de689fdb510972ef583f3ac/kyaml/internal/forked/github.com/go-yaml/yaml/yaml.go#L488)

Reply:
Yes, you are right. Updated